### PR TITLE
codegen: support polymorphism with actor inherit

### PR DESCRIPTION
**Pull request by goldenleaves**

### CODE REVIEW

File: include/hiactor/core/actor_message.hh [hunk 1]
```diff
@@ -59,9 +59,25 @@ struct address {
...
+    inline uint32_t get_shard_id() const {
         return load_unaligned_int<uint32_t>(data);
     }
+
+    // Set/Get the actor type id to which the target actor method really belongs.
+    //
+    // This is useful to support actor polymorphism and distinguish actor method calls
+    // of parent and inherited actors.
+    //
+    // The method actor type is stored in the tail of `addr`.
```
Comment (LiSu) [inline include/hiactor/core/actor_message.hh]: rephrase

File: src/core/actor-template.cc [hunk 2]
```diff
@@ -186,4 +186,138 @@ void actor_group::notify_child_stopped() {
     _actor_activation_sem.signal(1);
 }
 
+
+actor::actor(actor_base* exec_ctx, const byte_t* addr, bool reentrant)
+    : actor_base(exec_ctx, addr), _max_concurrency(reentrant? UINT32_MAX : 1) {
```
Comment (LiSu) [inline src/core/actor-template.cc]: space before ？

**Review by LiSu — APPROVED** [review]

File: include/hiactor/core/actor_message.hh (hunk 1 above)
Comment (goldenleaves) [reply include/hiactor/core/actor_message.hh]: done

File: src/core/actor-template.cc (hunk 2 above)
Comment (goldenleaves) [reply src/core/actor-template.cc]: ok

**Pull request merged**